client/services/FeedbackGeneratorService.js

```diff
@@ -30,6 +30,65 @@ tie.factory('FeedbackGeneratorService', [
     RUNTIME_ERROR_FEEDBACK_MESSAGES, WRONG_LANGUAGE_ERRORS, LANGUAGE_PYTHON,
     CLASS_NAME_AUXILIARY_CODE, CLASS_NAME_SYSTEM_CODE, PARAGRAPH_TYPE_TEXT,
     PARAGRAPH_TYPE_CODE, PARAGRAPH_TYPE_SYNTAX_ERROR) {
+
+    /**
+     * Global constant to act as the threshhold before prompting the user
+     * to look at the language primer.
+     *
+     * @type {number}
+     */
+    var UNFAMILIARITY_THRESHOLD = 5;
+
+    /**
+     * Global number to keep track of consecutive syntax errors.
```

Comment:
In what way is this global? It looks like it's scoped to this service.

Maybe s/Global number/Counter. Ditto elsewhere.